### ISSUE TITLE
Responsive breakpoints are overlapping and causing undesired behavior

### ISSUE THREAD

**Issue by petko-inev** [akiran/react-slick#777]
The breakpoints in the responsive array generate overlapping media queries.
In this example https://github.com/akiran/react-slick/blob/master/examples/Responsive.js the breakpoints are 480, 600 and 1024. If you open a browser with 480 screen width (or adjust it via devtools) and refresh the page you can see two items per slide in the "Responsive" example while there should be only 1, according to the breakpoints. 
I'm guessing this happens because the carousel settings are controlled by matchmedia listeners and both listeners are getting called. This would be expected since the media queries generated by https://github.com/akiran/react-slick/blob/master/src/slider.js#L32 would overlap. A possible solution would be to increase the lower boundary (min-width) of the match medias by 1 so that only one matchmedia listener would be called for a given screen size.

**Comment by petko-inev** [comment]
Tested this on Chrome, Safari and Firefox and the behavior is somewhat different. The smallest breakpoint (480) does work in Safari but at the next one (600) it displays 3 items per page (should be 2).
Bottom line I think that the overlapping media queries produce and unpredictable behavior.
I did a small fix locally that seems to resolve the issue - added +1 to minWidth here - https://github.com/akiran/react-slick/blob/master/src/slider.js#L37 , can do a PR if needed.
